include/grpcpp/server_builder.h

```diff
@@ -210,15 +210,44 @@ class ServerBuilder {
   /// doc/workarounds.md.
   ServerBuilder& EnableWorkaround(grpc_workaround_list id);
 
- private:
-  friend class ::grpc::testing::ServerBuilderPluginTest;
-
+ protected:
   struct Port {
     grpc::string addr;
     std::shared_ptr<ServerCredentials> creds;
     int* selected_port;
   };
 
+  typedef std::unique_ptr<grpc::string> HostString;
+  struct NamedService {
+    explicit NamedService(Service* s) : service(s) {}
+    NamedService(const grpc::string& h, Service* s)
+        : host(new grpc::string(h)), service(s) {}
+    HostString host;
+    Service* service;
+  };
+
+  std::vector<Port> ports() { return ports_; }
+
+  std::vector<std::reference_wrapper<NamedService>> services() {
+    std::vector<std::reference_wrapper<NamedService>> service_refs;
+    for (auto &ptr : services_) {
+      service_refs.push_back(std::ref(*ptr));
+    }
```

Comment:
So this is a vector of references to unique_ptr? Given that you're not transferring ownership, why not just have this be a vector of dumb pointers acquired through `->get` on each element of the iterator? I think that's more consistent with our overall style.